hugr-core/src/ops/custom.rs

```diff
@@ -392,9 +392,14 @@ pub fn resolve_opaque_op(
                 r.name().clone(),
             ));
         };
-        let ext_op =
-            ExtensionOp::new(def.clone(), opaque.args.clone(), extension_registry).unwrap();
-        if opaque.signature != ext_op.signature {
+        let ext_op = ExtensionOp::new(def.clone(), opaque.args.clone(), extension_registry)?;
```

Comment:
The change from `unwrap` to `?` here necessitates the addition of  `CustomOpError::SignatureError`? Good move :)